src/shared/watchdog.c

```diff
@@ -31,6 +31,7 @@
 #include "watchdog.h"
 
 static int watchdog_fd = -1;
+static const char *watchdog_dev = "/dev/watchdog";
```

Comment:
uh, i really don't like the memory management for this. This should either always point to static memory, or always to dynamic memory. But right now it initially points to static memory, except when the user overrides it, in which case it points to externally owned dynamic memory, which might go away any time.

Please let's clean up memory management here, let's stick one kind of memory: dynamic. Specifically, change "const char*" to just "char*", and initialize this to NULL. When open_watchdog() sees this set to NULL it should use "/dev/watchdog" instead, i.e. the fallbcak should happen at the moment we read the variable, not already at initialization.

Then, make sure that watchdog_set_device() allocates a new string, and make sure there's watchdog_free_device() or so, that frees the string, and that is called at PID 1 exit, so that we are valgrind clean. Yes, this means we'll keep the string in memory twice (once in main.c's arg_watchdog_device, and once in this copy), but that's fine. I very much prefer clean memory owenership rules about minimizing copies.

Reply:
Makes sense, will change, thanks.